### DISCUSSION TITLE
The consistency of result between text-based query and ORM-based query.

### DISCUSSION THREAD

**Discussion by jiseongg** [sqlalchemy/sqlalchemy#11203] [Usage Questions] [answered]
Hi, there. 

I am developing a software layer to manage database using SQLAlchemy. The library relies heavily on SQLAlchemy ORM. Me and my colleagues prefer using ORM to using text query, but there are several situations we have to retrieve data with complex query, which leads us to develop API based on text-based query.

Here is what I am wondering. Consider the ORM-mapped class `ORM_A` which is mapped to the table `TB_A`. We can fetch all data from `TB_A` in the following two ways.
```python
# using ORM
from sqlalchemy import select
session.execute(select(ORM_A)).fetchall()

# using text query
from sqlalchemy.sql.expression import text
session.execute(text("SELECT * FROM TB_A")).fetchall()
```

I found that through the result of first one we can use the instance of `ORM_A`. However, the result of the second one doesn't seem to return `ORM_A` instance. I guess that the first one helps SQLAlchemy to map `Row` object to pre-loaded ORM but the second doesn't. Aren't there any simple way to post-process the second result to construct set of `ORM_A` instances?

Thanks.

**Comment by CaselIT** [comment, marked as answer]
Hi,
You can use `select(ORM_A).from_statement(text(...))`  See https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.Select.from_statement

**Reply by jiseongg** [reply]
Thank you for the answer. I hope (and feel quite sure) this method will help us support the both ways seamlessly.
